crates/epaint/src/image.rs

```diff
@@ -120,6 +120,16 @@ impl ColorImage {
         Self { size, pixels }
     }
 
+    /// Alternative method to `from_gray`.
+    /// Create a [`ColorImage`] from iterator over flat opaque gray data.
+    /// 
+    /// Panics if `size[0] * size[1] != gray_iter.len()`.
+    pub fn from_gray_iter<'_>(size: [usize; 2], gray_iter: impl Iterator<Item = &'_ u8> + ExactSizeIterator) -> Self {
+        assert_eq!(size[0] * size[1], gray_iter.len());
+        let pixels = gray_iter.map(|p| Color32::from_gray(*p)).collect();
```

Comment:
```suggestion
    pub fn from_gray_iter(size: [usize; 2], gray_iter: impl Iterator<Item = u8>) -> Self {
        let pixels = gray_iter.map(Color32::from_gray).collect();
        assert_eq!(size[0] * size[1], pixels.len());
```

Reply:
This will make it more general

Reply:
Absolutely agree. But the pull request is still here, maybe it was caused by my wrong click on the button "Commit Suggestion".